ibis/backends/postgres/__init__.py

```diff
@@ -144,6 +145,9 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
 
             columns = schema.keys()
             df = op.data.to_frame()
+            # nan gets compiled into 'NaN'::float which throws errors in non-float columns
+            df = df.replace(np.nan, None)
+
```

Comment:
Can we limit this replacement to columns that aren't already floating point?  I think replacing a pandas NaN with `NULL` makes sense for non-float columns, but I'm wary of doing it across the board